### PR TITLE
Tax implementation In Netsuite

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

**Review by Shwetabhk — APPROVED** [review]

File: src/app/netsuite/settings/netsuite-configurations/configuration/configuration.component.html [hunk 1]
```diff
@@ -113,6 +113,10 @@
         <div class="schedule--toggle-text">Import NetSuite Categories / Accounts into Fyle</div>
       </mat-slide-toggle>
 
+      <mat-slide-toggle color="primary" class="schedule" formControlName="importTaxDetails">
+        <div class="schedule--toggle-text">Import Netsuite Tax Details to Fyle</div>
```
Comment (ashwin1111) [inline src/app/netsuite/settings/netsuite-configurations/configuration/configuration.component.html:117]: NetSuite

File: src/app/netsuite/settings/netsuite-configurations/expense-field-configuration/expense-field-configuration.component.ts [hunk 2]
```diff
@@ -177,7 +177,7 @@ export class ExpenseFieldConfigurationComponent implements OnInit {
     const that = this;
 
     that.mappingSettings = mappingSetting.filter(
-      setting => setting.source_field !== 'EMPLOYEE' && setting.source_field !== 'CATEGORY'
+      setting => setting.source_field !== 'EMPLOYEE' && setting.source_field !== 'CATEGORY' && setting.source_field !== 'TAX_GROUP'
```
Comment (ashwin1111) [inline src/app/netsuite/settings/netsuite-configurations/expense-field-configuration/expense-field-configuration.component.ts:180]: Add this in API

**Review by ashwin1111 — CHANGES_REQUESTED** [review]

File: src/app/core/models/general-setting.model.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ export type GeneralSetting = {
     reimbursable_expenses_object: string;
     corporate_credit_card_expenses_object: string;
     import_projects?: boolean;
+    import_tax_items?: boolean;
```
Comment (ashwin1111) [inline src/app/core/models/general-setting.model.ts:7]: ? not needed

File: src/app/netsuite/settings/netsuite-configurations/expense-field-configuration/expense-field-configuration.component.ts [hunk 4]
```diff
@@ -178,7 +178,7 @@ export class ExpenseFieldConfigurationComponent implements OnInit {
...
```
Comment (ashwin1111) [inline src/app/netsuite/settings/netsuite-configurations/expense-field-configuration/expense-field-configuration.component.ts:181]: this is not needed right?

File: src/app/core/services/settings.service.ts [hunk 5]
```diff
@@ -105,4 +107,22 @@ export class SettingsService {
...
+  postSubsidiaryMappings(subsidiaryMappingPayload: SubsidiaryMapping = null): Observable<SubsidiaryMapping> {
+    const workspaceId = this.workspaceService.getWorkspaceId();
+
+    return this.apiService.post(`/workspaces/${workspaceId}/mappings/subsidiaries/`, subsidiaryMappingPayload);
+  }
+
+  @CacheBuster({
+    cacheBusterNotifier: subsidiaryMappingCache$
+  })
+  postCountryDetails(): Observable<SubsidiaryMapping> {
```
Comment (ashwin1111) [inline src/app/core/services/settings.service.ts:123]: Will suggest something here

**Review by ashwin1111 — APPROVED** [review]

File: src/app/netsuite/settings/netsuite-configurations/expense-field-configuration/expense-field-configuration.component.ts (hunk 4 above)
Comment (NileshPant1999) [reply src/app/netsuite/settings/netsuite-configurations/expense-field-configuration/expense-field-configuration.component.ts:181]: yes

**Review by Sravanksk — APPROVED** [review]

**Pull request merged**